arkane/encorr/reference.py

```diff
@@ -665,5 +669,57 @@ def _is_valid_calculated_data(data_dictionary):
     return False
 
 
+def update_reference_db(path: str, model_chemistry: str = None, ref_paths: List[str] = None) -> None:
+    """
+    Update the reference species database from a collection of ArkaneSpecies YAML files. All of the YAML files should
+    be included in a single folder.
+
+    Notes:
+        ArkaneSpecies YAML files must end in either '.yml' or '.yaml'
+
+    Args:
+        path (str): The path to a directory containing all of the ArkaneSpecies YAML files
+        model_chemistry (str): Model chemistry string matching the Arkane format to be updated. If not supplied will
+            default to the ``level_of_theory`` parameter of the ArkaneSpecies objects
+        ref_paths (list): A list to the paths of reference sets to load. By default only the main reference set is
+            loaded
+
+    Returns:
+        None
+    """
+    spcs_list = []
+    for f in os.listdir(path):
+        if any(file_extension in f.lower() for file_extension in ['.yml', '.yaml']):
+            spc = ArkaneSpecies.__new__(ArkaneSpecies)
```

Comment:
This isn't so much an issue with this PR but more of an issue with the `ArkaneSpecies` class. Initializing an empty instance in this way seems a little strange. I feel like there should be a mechanism to just initialize an instance by providing a yaml file directly.

Reply:
I agree. Part of the issue is that you can't initialize an ArkaneSpecies object without passing arguments to __init__. Having a separate function for this feels clunky as well. I wonder if we can get around this by having `load_yaml` return the ArkaneSpecies object, so that this method can be called on the class rather than an instance. I'll investigate and if it works I'll push a commit